src/Microsoft.AspNetCore.Routing/Constraints/StringRouteConstraint.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Routing.Constraints
+{
+    /// <summary>
+    /// Constrains a route parameter to contain only a specifyed strign.
+    /// </summary>
+    public class StringRouteConstraint : IRouteConstraint
+    {
+        private string _value;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="StringRouteConstraint"/> class.
+        /// </summary>
+        /// <param name="value">The constraint value to match.</param>
+        public StringRouteConstraint(string value)
+        {
+            if (value == null)
+            {
+                throw new ArgumentNullException(nameof(value));
+            }
+
+            _value = value;
+        }
+
+        /// <inheritdoc />
+        public bool Match(HttpContext httpContext, IRouter route, string routeKey, RouteValueDictionary values, RouteDirection routeDirection)
+        {
+            if (httpContext == null)
+            {
+                throw new ArgumentNullException(nameof(httpContext));
+            }
+
+            if (route == null)
+            {
+                throw new ArgumentNullException(nameof(route));
+            }
+
+            if (routeKey == null)
+            {
+                throw new ArgumentNullException(nameof(routeKey));
+            }
+
+            if (values == null)
+            {
+                throw new ArgumentNullException(nameof(values));
+            }
+
+            object routeValue;
+
+            if (values.TryGetValue(routeKey, out routeValue)
+                && routeValue != null)
+            {
+                var parameterValueString = Convert.ToString(routeValue, CultureInfo.InvariantCulture);
+
```

Comment:
the if statement could be modified

```
            var routeValue = Convert.ToString(values[routeKey], CultureInfo.InvariantCulture);

            if (routeValue != null)
            {
                return routeValue.Equals(_value, StringComparison.OrdinalIgnoreCase);
            }
```
because the RouteValueDictionary makes the control of the key presence, but did not do it for consistency with the other constraints.